tools/nuget/Desktop/Microsoft.MixedReality.WebRTC.props

```diff
@@ -0,0 +1,9 @@
+﻿<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Condition="'$(UsingMicrosoftNETSdk)' != 'true'">
+    <Content Include="$(MSBuildThisFileDirectory)\..\runtimes\win10-$(PlatformTarget)\native\mrwebrtc.dll">
```

Comment:
From memory`$(PlatformTarget)` is defined after the `props` files are included, it's usually defined in a `PropertyGroup` based on `$(Platform)`. Perhaps this works on newer versions of `VS`, I'd be surprised if it worked on older versions.

I expect you'll run into the same issue as described in https://github.com/dotnet/project-system/issues/4158#issuecomment-645671777 where the file copy doesn't happen on a clean checkout as the `.props` file doesn't exist on disk and isn't evaluated as part of the project when using older `NonSdkStyle` project (packages.config).

> When the restore occurs the files will be written to disk. If we were to re-evaluate the project at that point we would > pick them up and the <Reference> items they contain. However, since we don't even know the <Import> exists we > don't know we need to re-evaluate. If we re-evaluate the project for some other reason (e.g., it is unloaded and > reloaded, or a .cs file or reference is added) we will then see the imported files.

I'd expect this to only work after closing/reopening the project/solution.

Reply:
@amaitland so if I understand you correctly there is no proper way to do this in the old project system apart from manually adding a props file similar to this to the project that includes this package?

Reply:
Though I guess is a "known issue" with the old project system, any NuGet package can add `.targets` or `.props` files that would suffer from the same problem.

Reply:
If you switch from a .props file to a .targets file the PlatformTarget will be accessible. 

I'm not aware of a perfect solution unfortunately. VS2017 will fail to load the props after a Nuget install/restore.  VS2013/2015 don't have the problem. VS2019 appears to be working, I had a few instances where the files weren't copied, might have been my fault as I was testing some changes, sample size of one so might be nothing. 

I've added an after build task with a condition to check if the file exists in the output folder, display a message in the build output, I was considering outputting an error so the build fails. You could do something similar. 

You could add a after build target to copy the file to the output directory, you'd probably need an after publish target also.

This is just from my experience, I'm not a definitive authority on the subject, just someone who's spent a lot of time hacking Nuget packages to try get things working. If someone has some other suggestions I'd be very interested in hearing them.

Reply:
I just tried creating a new .NET Framework console app (4.7.2) and adding a manually-generated package with the config in this PR, after fixing the path (since the `.props` file is in `build\netstandard2.0`, so 2 folders down from the root, not 1):

```diff
- <Content Include="$(MSBuildThisFileDirectory)\..\runtimes\win10-$(PlatformTarget)\native\mrwebrtc.dll">
+ <Content Include="$(MSBuildThisFileDirectory)\..\..\runtimes\win10-$(PlatformTarget)\native\mrwebrtc.dll">
```

(ping @sanderaernouts can you fix the PR please?)

...and the only issue I got is that by default `$(PlatformTarget)` is `AnyCPU` because VS only creates that target. That error _will_ fail the build and be rather self-explanatory I think. After manually adding an `x64` target (for example) and switching to that, the project builds and runs without issue. I am using Visual Studio 2019 16.8.

@amaitland noted, thanks for the detailed explanation. But at this point we don't really support VS 2017 officially, that would make a bit too much work to test also all VS versions, so we can only guarantee VS 2019. So I think we can ignore that issue for now to unblock this PR. That being said after that if you want to propose a change to add an after build task for VS 2017 I am happy to have a look at it.

Reply:
@djee-ms I have updated the paths in the props files to go 2 levels up instead of 1.

> ...and the only issue I got is that by default $(PlatformTarget) is AnyCPU because VS only creates that target. That error will fail the build and be rather self-explanatory I think. After manually adding an x64 target (for example) and switching to that, the project builds and runs without issue. I am using Visual Studio 2019 16.8.

regarding this ☝️ , we could define a default when AnyCPU is used, either `x86` or `x64` but I don't know if that is correct? Another pattern I have seen with `AnyCPU` and native code is to include both versions in `x64` and `x86` subfolders but then the code that is invoking `mrwebrtc.dll` needs to decide on using `x64\mrwebrtc.dll` or `x86\mrwebrtc.dll` based on the actual platform it is running on.